### PR TITLE
Immutable js root state

### PR DESCRIPTION
**Pull request by fabriziomoscon**
Addresses issue #37 

This patch allows to use `redux-offline` with a root immutable state.
These are non breaking changes on top of `v2.1.0`

- added config `offlineStateLens`: with state getter and setter and splitter for `offline` state vs rest
- added config `persistAutoRehydrate`: to inject another autoHydration (eg: `redux-persist-immutable`)
- used `offlineStateLens` in the middleware and the enhancer
- added the new flow types for the config
- fixed `config.persist` flow type
- updated the README.md

An auxiliary npm module providing immutable default configs has been published at: https://www.npmjs.com/package/redux-offline-immutable-config

This code has been written together with @Ashmalech

### CODE REVIEW

**Comment by wacii** [comment]
Thanks for the PR!

Made a couple small changes to the README. Also we changed how the store enhancer is applied, so I reverted that.

**Comment by wacii** [comment]
This needs to be merged into develop. I'll look into resolving the conflicts later this week, unless you want to rebase it again.

**Comment by fabriziomoscon** [comment]
I can try rebasing, I will let you know if I get stuck

File: src/index.js [hunk 1]
```diff
@@ -40,15 +39,15 @@ export const offline = (userConfig: $Shape<Config> = {}) => (
...
+  const offlineReducer = enhanceReducer(reducer, config);
 
   const offlineMiddleware = applyMiddleware(createOfflineMiddleware(config));
 
   // create autoRehydrate enhancer if required
   const offlineEnhancer =
-    config.persist && config.rehydrate
-      ? compose(offlineMiddleware, autoRehydrate())
-      : offlineMiddleware;
+    config.persist && config.rehydrate && config.persistAutoRehydrate
```
Comment (fabriziomoscon) [inline src/index.js]: @wacii Please check that I kept the correct code for this block during the rebase

**Comment by fabriziomoscon** [comment]
@wacii I have rebased on top of `develop` and run `yarn run prettier`. I had one comment for a single change, but the rest was merged automatically by git.
Let me know if you need anything else.

**Comment by wacii** [comment]
Nifty. I addressed your comment. Everything looks good.

Thanks again.

**Pull request merged**